### PR TITLE
Bugfix : 44761

### PR DESCRIPTION
**Pull request by say1bt**
Removed Nested button
Fixed the issue :  when user selects the date in default value tab it shows previous day's date

### CODE REVIEW

File: ValidationRules/src/Features/index.tsx [hunk 1]
```diff
@@ -1734,14 +1734,14 @@ const ParentComponent = ({
                       >
                         {"+ " + languageConstants?.ExpressionBuilder_AddButton}
                       </Button>
-                      <Button
+                      {/* <Button
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/index.tsx:1737]: why are you comment this button for fixed this issue

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 2]
```diff
@@ -329,38 +378,38 @@ function SectionContainer({
...
+          },
         ])
       );
     }
   };
 
   const toggleEnableOnClickMin = () => {
     setToggledEnableMin(!toggleEnableMin);
-    setMinValue({ input: '', changedId: '', fieldName: 'minValue' });
+    setMinValue({ input: "", changedId: "", fieldName: "minValue" });
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:390]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 3]
```diff
@@ -329,38 +378,38 @@ function SectionContainer({
...
     handleMinMaxWhenToggleChanged(null, null);
   };
 
   const toggleEnableOnClickMax = () => {
     setToggledEnableMax(!toggleEnableMax);
-    setMaxValue({ input: '', changedId: '', fieldName: 'maxValue' });
+    setMaxValue({ input: "", changedId: "", fieldName: "maxValue" });
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:396]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 4]
```diff
@@ -329,38 +378,38 @@ function SectionContainer({
...
     handleMinMaxWhenToggleChanged(null, null);
   };
 
   const onChangeCheckBoxMin = (e: any) => {
     setMinCheckboxEnabled(e.target.checked);
     setMinMaxCheckboxEnabled((prev: any) => {
-      return { ...prev, ['minCheckbox']: e.target.checked };
+      return { ...prev, ["minCheckbox"]: e.target.checked };
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:403]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 5]
```diff
@@ -329,38 +378,38 @@ function SectionContainer({
...
     });
 
     if (!e.target.checked) {
-      setMinValue({ input: '', changedId: '', fieldName: 'minValue' });
+      setMinValue({ input: "", changedId: "", fieldName: "minValue" });
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:407]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 6]
```diff
@@ -329,38 +378,38 @@ function SectionContainer({
...
       handleMinMaxWhenToggleChanged(null, null);
     } else {
       if (!minValue?.input) {
         setValidation((prev: any) => {
-          return { ...prev, ['minMaxValidation']: false };
+          return { ...prev, ["minMaxValidation"]: false };
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:412]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 7]
```diff
@@ -369,15 +418,15 @@ function SectionContainer({
   const onChangeCheckBoxMax = (e: any) => {
     setMaxCheckboxEnabled(e.target.checked);
     setMinMaxCheckboxEnabled((prev: any) => {
-      return { ...prev, ['maxCheckbox']: e.target.checked };
+      return { ...prev, ["maxCheckbox"]: e.target.checked };
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:421]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 8]
```diff
@@ -369,15 +418,15 @@ function SectionContainer({
...
     });
     if (!e.target.checked) {
-      setMaxValue({ input: '', changedId: '', fieldName: 'maxValue' });
+      setMaxValue({ input: "", changedId: "", fieldName: "maxValue" });
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:424]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 9]
```diff
@@ -369,15 +418,15 @@ function SectionContainer({
...
       handleMinMaxWhenToggleChanged(null, null);
     } else {
       if (!maxValue?.input) {
         setValidation((prev: any) => {
-          return { ...prev, ['minMaxValidation']: false };
+          return { ...prev, ["minMaxValidation"]: false };
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:429]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 10]
```diff
@@ -404,7 +455,7 @@ function SectionContainer({
   for (let i = 10; i < 36; i++) {
     options.push({
       label: i.toString(36) + i,
-      value: i.toString(36) + i
+      value: i.toString(36) + i,
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:458]: put meaningful name to the variables without putting "x" or something

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 11]
```diff
@@ -436,64 +489,68 @@ function SectionContainer({
...
-            value: addValue
-          }
+            type: "ADD_V",
+            value: addValue,
+          },
         ])
       );
-    } else if (radioDefaultValOption === 'VAL_Q') {
-      console.log('Add Value Questions ');
+    } else if (radioDefaultValOption === "VAL_Q") {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:512]: get hard coded values from constant file
remove consoles

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 12]
```diff
@@ -436,64 +489,68 @@ function SectionContainer({
...
+      console.log("Add Value Questions ");
       _setNestedRows(
         updateAllLevelActionsArray(_nestedRows, sectionLevel, [
           {
-            type: 'VAL_Q',
-            value: addValue
-          }
+            type: "VAL_Q",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:517]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 13]
```diff
@@ -436,64 +489,68 @@ function SectionContainer({
...
+            value: addValue,
+          },
         ])
       );
-    } else if (radioDefaultValOption === 'MAT_F' && addValue?.length) {
-      console.log('Add Value Questions ');
+    } else if (radioDefaultValOption === "MAT_F" && addValue?.length) {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:522]: get hard coded values from constant file
remove consoles

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 14]
```diff
@@ -436,64 +489,68 @@ function SectionContainer({
...
+      console.log("Add Value Questions ");
       _setNestedRows(
         updateAllLevelActionsArray(_nestedRows, sectionLevel, [
           {
-            type: 'MAT_F',
-            value: addValue
-          }
+            type: "MAT_F",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:527]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 15]
```diff
@@ -436,64 +489,68 @@ function SectionContainer({
...
       );
     } else {
       if (radioDefaultValOption) {
         _setNestedRows(
           updateAllLevelActionsArray(_nestedRows, sectionLevel, [
             {
-              type: 'MAT_F',
-              value: addValue
-            }
+              type: "MAT_F",
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:537]: get hard coded values from constant file

File: ValidationRules/src/Features/index.tsx (hunk 1 above)
Comment (say1bt) [reply ValidationRules/src/Features/index.tsx:1737]: This is to remove the top level Add Nested Button

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 16]
```diff
@@ -436,64 +489,68 @@ function SectionContainer({
...
+            },
           ])
         );
       }
     }
   }, [radioDefaultValOption, addValue]);
 
   useEffect(() => {
-    if (radioDefaultValOption === 'MAT_F' && checkedReferences) {
+    if (radioDefaultValOption === "MAT_F" && checkedReferences) {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:547]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 17]
```diff
@@ -503,6 +560,24 @@ function SectionContainer({
     }
   }, [pickOperatorDefault, pickQuestionDefault, pickQuestion2Default]);
 
+  useEffect(() => {
+    _nestedRows.forEach((item: any) => {
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:564]: write this function as a common function and call it inside useEffect

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 18]
```diff
@@ -503,6 +560,24 @@ function SectionContainer({
...
+            const formattedDate: any = utcDate.local().format("YYYY-MM-DD");
+            action.value = formattedDate;
+            setAddValue(formattedDate);
+          }
+        });
+      });
+    });
+  }, []);
+
+  console.log("_nestedRows", _nestedRows);
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:579]: remove consoles

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 19]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
             </div>
           </div>
           {currentPossitionDetails &&
-            currentPossitionDetails?.currentPosition === 'question' &&
-            currentQuestionDetails?.questionType !== 'Header' &&
-            currentQuestionDetails?.questionType !== 'List' &&
-            currentQuestionDetails?.questionType !== 'Date' && (
+            currentPossitionDetails?.currentPosition === "question" &&
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:632]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 20]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
+                          suerveyIsPublished
+                            ? suerveyIsPublished
+                            : !minCheckboxEnabled
+                        }
                         defaultChecked={
-                          typeof _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]
-                            ?.actions[0]?.minMax?.minValue !== 'string' ||
-                          _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]?.actions[0]
-                            ?.minMax?.minValue === '0'
+                          typeof _nestedRows?.find(
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:671]: put meaningful name to the variables without putting "x" or something
and use common function without re coding

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 21]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
+                          }`
+                        : `${languageConstants?.ExpressionBuilder_MinLengthConstants}` +
+                          ":"}
                     </div>
                     {toggleEnableMin ? (
                       <NumberInputField
                         selectedValue={
-                          _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]?.actions[0]
-                            ?.minMax?.minValue
+                          _nestedRows?.find((x: any) => x[sectionLevel])?.[
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:694]: put meaningful name to the variables without putting "x" or something
and use common function without re coding

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 22]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
                         defaultDisabled={
-                          suerveyIsPublished ? suerveyIsPublished : !minCheckboxEnabled
+                          suerveyIsPublished
+                            ? suerveyIsPublished
+                            : !minCheckboxEnabled
                         }
                         setInputNumber={setMinValue}
                         changedId={undefined}
-                        fieldName={'minValue'}
+                        fieldName={"minValue"}
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:706]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 23]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
                         validatingSuccess={minMaxValidation}
                       />
                     ) : (
                       <DropDown
                         dropDownData={
                           questionList && questionList?.length
-                            ? questionList.filter((ques: any) => ques?.questionType === 'Numeric')
+                            ? questionList.filter(
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:713]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 24]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
+                            : !minCheckboxEnabled
+                        }
                         setExpression={setMinValue}
                         changedId={undefined}
-                        fieldName={'minValue'}
+                        fieldName={"minValue"}
                         selectedValue={
-                          _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]?.actions[0]
-                            ?.minMax?.minValue || ''
+                          _nestedRows?.find((x: any) => x[sectionLevel])?.[
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:727]: put meaningful name to the variables without putting "x" or something
and use common function without re coding

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 25]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
+                          suerveyIsPublished
+                            ? suerveyIsPublished
+                            : !maxCheckboxEnabled
+                        }
                         defaultChecked={
-                          typeof _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]
-                            ?.actions[0]?.minMax?.maxValue !== 'string' ||
-                          _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]?.actions[0]
-                            ?.minMax?.maxValue === '0'
+                          typeof _nestedRows?.find(
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:762]: put meaningful name to the variables without putting "x" or something
and use common function without re coding

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 26]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
-                        : `${languageConstants?.ExpressionBuilder_MaxLengthConstants}` + ' :'}
+                        : `${languageConstants?.ExpressionBuilder_MaxLengthConstants}` +
+                          " :"}
                     </div>
                     {toggleEnableMax ? (
                       <NumberInputField
                         selectedValue={
-                          _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]?.actions[0]
-                            ?.minMax?.maxValue || ''
+                          _nestedRows?.find((x: any) => x[sectionLevel])?.[
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:782]: put meaningful name to the variables without putting "x" or something
and use common function without re coding

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 27]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
                         defaultDisabled={
-                          suerveyIsPublished ? suerveyIsPublished : !maxCheckboxEnabled
+                          suerveyIsPublished
+                            ? suerveyIsPublished
+                            : !maxCheckboxEnabled
                         }
                         setInputNumber={setMaxValue}
                         changedId={undefined}
-                        fieldName={'maxValue'}
+                        fieldName={"maxValue"}
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:794]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 28]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
                         validatingSuccess={minMaxValidation}
                       />
                     ) : (
                       <DropDown
                         dropDownData={
                           questionList && questionList?.length
-                            ? questionList.filter((ques: any) => ques?.questionType === 'Numeric')
+                            ? questionList.filter(
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:801]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 29]
```diff
@@ -552,147 +629,192 @@ function SectionContainer({
...
+                            : !maxCheckboxEnabled
+                        }
                         setExpression={setMaxValue}
                         changedId={undefined}
-                        fieldName={'maxValue'}
+                        fieldName={"maxValue"}
                         selectedValue={
-                          _nestedRows?.find((x: any) => x[sectionLevel])?.[sectionLevel]?.actions[0]
-                            ?.minMax?.maxValue || ''
+                          _nestedRows?.find((x: any) => x[sectionLevel])?.[
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:815]: put meaningful name to the variables without putting "x" or something
and use common function without re coding

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 30]
```diff
@@ -707,12 +829,14 @@ function SectionContainer({
...
               <div className="referneces-acts">
                 <div className="mr-10">
-                  {languageConstants?.ExpressionBuilder_AddQuesRef + ' :'}
+                  {languageConstants?.ExpressionBuilder_AddQuesRef + " :"}
                 </div>
                 <div>
                   <Switch
                     defaultChecked={
-                      defaultActionSetWhenRetriving?.type === 'disable' ? false : true
+                      defaultActionSetWhenRetriving?.type === "disable"
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:837]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 31]
```diff
@@ -723,50 +847,59 @@ function SectionContainer({
...
                   value={checkedReferences ? radioDefaultValOption : null}
-                  disabled={suerveyIsPublished ? suerveyIsPublished : !checkedReferences}
+                  disabled={
+                    suerveyIsPublished ? suerveyIsPublished : !checkedReferences
+                  }
                   defaultValue={defaultActionSetWhenRetriving?.type}
                 >
-                  <Radio value={'CLE_Q'}>{languageConstants?.ExpressionBuilder_ClrQues}</Radio>
-                  <Radio value={'ADD_V'}>
+                  <Radio value={"CLE_Q"}>
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:855]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 32]
```diff
@@ -723,50 +847,59 @@ function SectionContainer({
...
+                    {languageConstants?.ExpressionBuilder_ClrQues}
+                  </Radio>
+                  <Radio value={"ADD_V"}>
                     {languageConstants?.ExpressionBuilder_AddValOrSetVal}
                   </Radio>
 
                   {currentQuestionDetails?.questionType !==
                     dbConstants?.questionTypes?.listQuestion && (
-                    <Radio value={'VAL_Q'}>
+                    <Radio value={"VAL_Q"}>
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:864]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 33]
```diff
@@ -723,50 +847,59 @@ function SectionContainer({
...
                       {languageConstants?.ExpressionBuilder_ValFromAnotherQues}
                     </Radio>
                   )}
 
                   {currentQuestionDetails?.questionType ===
                     dbConstants?.questionTypes?.numericQuestion && (
-                    <Radio value={'MAT_F'}>
+                    <Radio value={"MAT_F"}>
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:871]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 34]
```diff
@@ -723,50 +847,59 @@ function SectionContainer({
...
                       {languageConstants?.ExpressionBuilder_MatheFormula}
                     </Radio>
                   )}
                 </Radio.Group>
               </div>
               <div className="default-options">
-                {radioDefaultValOption === 'MAT_F' ? (
+                {radioDefaultValOption === "MAT_F" ? (
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:878]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 35]
```diff
@@ -806,25 +943,25 @@ function SectionContainer({
                         <div className="exp-input-wrap">
                           <div
                             className="num-input"
-                            onClick={() => handleMathematicalOperator('+')}
+                            onClick={() => handleMathematicalOperator("+")}
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:946]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 36]
```diff
@@ -881,36 +1026,44 @@ function SectionContainer({
...
-                      style={{ width: '50%' }}
+                      style={{ width: "50%" }}
                       onChange={(e: any) => {
                         setAddValue(e);
                       }}
                       defaultValue={defaultActionSetWhenRetriving?.value}
                       value={addValue}
                     />
-                  ) : currentQuestionDetails?.questionType === 'Date' ? (
+                  ) : currentQuestionDetails?.questionType === "Date" ? (
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1066]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 37]
```diff
@@ -926,21 +1079,23 @@ function SectionContainer({
...
-                            : dayjs(moment().format(dbConstants?.common?.dateFormat))
+                            : dayjs(
+                                moment().format(dbConstants?.common?.dateFormat)
+                              )
                         }
                         format={dbConstants?.common?.dateFormat}
                         // disabled={isDisabled}
                         onChange={(input, option) => setAddValue(input)}
-                        style={{ width: '150px' }}
+                        style={{ width: "150px" }}
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1089]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 38]
```diff
@@ -926,21 +1079,23 @@ function SectionContainer({
...
                       />
                     </Space>
-                  ) : currentQuestionDetails?.questionType === 'List' ? (
+                  ) : currentQuestionDetails?.questionType === "List" ? (
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1092]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 39]
```diff
@@ -955,19 +1110,19 @@ function SectionContainer({
...
                       disabled={suerveyIsPublished}
                       placeholder={languageConstants?.addValue}
-                      style={{ width: '50%' }}
+                      style={{ width: "50%" }}
                       onChange={(e: any) => {
                         setAddValue(e?.target?.value);
                       }}
                       defaultValue={
-                        defaultActionSetWhenRetriving?.type === 'ADD_V'
+                        defaultActionSetWhenRetriving?.type === "ADD_V"
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1118]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 40]
```diff
@@ -955,19 +1110,19 @@ function SectionContainer({
...
                           ? defaultActionSetWhenRetriving?.value
                           : null
                       }
                       value={addValue}
                     />
                   )
-                ) : radioDefaultValOption === 'VAL_Q' ? (
+                ) : radioDefaultValOption === "VAL_Q" ? (
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1125]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 41]
```diff
@@ -982,27 +1137,32 @@ function SectionContainer({
                         //     ? defaultActionSetWhenRetriving?.value
                         //     : null
                         // }
-                        style={{ width: '50%' }}
+                        style={{ width: "50%" }}
                         onChange={(e: any) => setAddValue(e)}
                         disabled={suerveyIsPublished}
                         options={
-                          currentQuestionDetails?.questionType === 'Numeric'
+                          currentQuestionDetails?.questionType === "Numeric"
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1144]: get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 42]
```diff
@@ -982,27 +1137,32 @@ function SectionContainer({
...
                             ? questionList?.filter(
                                 (x: any) =>
-                                  x['questionType'] ===
-                                    dbConstants?.questionTypes?.numericQuestion &&
-                                  x['questionType'] !== dbConstants?.questionTypes?.gridQuestion
+                                  x["questionType"] ===
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1147]: put meaningful name to the variables without putting "x" or something
and use common function without re coding
get hard coded values from constant file

File: ValidationRules/src/Features/sectionContainer.tsx [hunk 43]
```diff
@@ -982,27 +1137,32 @@ function SectionContainer({
...
+                                    dbConstants?.questionTypes?.gridQuestion
                               )
-                            : currentQuestionDetails?.questionType === 'Date'
+                            : currentQuestionDetails?.questionType === "Date"
                             ? questionList?.filter(
                                 (x: any) =>
-                                  x['questionType'] ===
-                                    dbConstants?.questionTypes?.dateTimeQuestion &&
-                                  x['questionType'] !== dbConstants?.questionTypes?.gridQuestion
+                                  x["questionType"] ===
```
Comment (tilangapramithvidusara) [inline ValidationRules/src/Features/sectionContainer.tsx:1156]: put meaningful name to the variables without putting "x" or something
and use common function without re coding
get hard coded values from constant file

**Review by tilangapramithvidusara — APPROVED** [review]

**Review by tilangapramithvidusara — APPROVED** [review]

**Review by tilangapramithvidusara — APPROVED** [review]

**Pull request merged**